test/test/basic-channel.js

```diff
@@ -0,0 +1,96 @@
+const Web3 = require('web3');
+
+const BigNumber = require('bignumber.js');
+
+const { expect } = require("chai")
+  .use(require("chai-as-promised"))
+  .use(require("chai-bignumber")(BigNumber));
+
+const {
+  bootstrap,
+  polkadotSenderSS58Alice,
+  polkadotSenderSS58Bob,
+  polkadotSenderSS58Charlie,
+  polkadotSenderSS58Dave,
+  polkadotSenderSS58Eve,
+  polkadotSenderSS58Ferdie,
+} = require('../src/fixtures');
+
+const { ChannelId } = require("../src/helpers");
+
+describe('Bridge', function () {
+  let ethClient, subClient;
+
+  before(async function () {
+    const clients = await bootstrap();
+    ethClient = clients.ethClient;
+    subClient = clients.subClient;
+    this.testParaEthAssetId = 0;
+  });
+
+  describe('Basic Channel', function () {
+    it('should transfer DOT for 6 accounts from Substrate to Ethereum (basic channel)', async function () {
+      const amount = BigNumber('10e+12'); // 10 DOT (12 decimal places in this environment)
+      const amountWrapped = BigNumber(Web3.utils.toWei('10', "ether")); // 10 SnowDOT (18 decimal places)
+      const ethAccount = ethClient.accounts[1];
+
+      const sudoKey = await subClient.api.query.sudo.key();
+      const sudoPair = subClient.keyring.getPair(sudoKey.toString());
+
+      const beforeEthBalance = await ethClient.getDotBalance(ethAccount);
+      const beforeSubBalanceAlice = await subClient.queryAccountBalance(polkadotSenderSS58Alice);
+      const beforeSubBalanceBob = await subClient.queryAccountBalance(polkadotSenderSS58Bob);
+      const beforeSubBalanceCharlie = await subClient.queryAccountBalance(polkadotSenderSS58Charlie);
+      const beforeSubBalanceDave = await subClient.queryAccountBalance(polkadotSenderSS58Dave);
+      const beforeSubBalanceEve = await subClient.queryAccountBalance(polkadotSenderSS58Eve);
+      const beforeSubBalanceFerdie = await subClient.queryAccountBalance(polkadotSenderSS58Ferdie);
+
+      // lock DOT using basic channel
+      await subClient.api.tx.sudo.sudo(
+        subClient.api.tx.utility.batchAll([
+          subClient.api.tx.utility.dispatchAs(
+            {system: {signed: subClient.alice.address}},
```

Comment:
Luckily found an example of this in an [Acala repo](https://github.com/AcalaNetwork/bodhi.js/blob/930cb66af4b5f94978c683dd390fb18778f720ac/scripts/virtual-tx-test.ts#L59)

Reply:
Nice! yeah I also search for code all the time on github to help with undocumented APIs